operator/src/test/java/oracle/kubernetes/operator/PodWatcherTest.java

```diff
@@ -379,6 +381,24 @@ void whenPodNotDeletedOnFirstRead_dontRunNextStep() {
     }
   }
 
+  @Test
+  void whenPodDeletedOnSecondRead_runNextStepOnlyOnce() {
+    AtomicBoolean stopping = new AtomicBoolean(false);
+    PodWatcher watcher = createWatcher(stopping);
+
+    testSupport.defineResources(createPod());
+    try {
+      testSupport.runSteps(watcher.waitForDelete(createPod(), terminalStep));
+      testSupport.failOnResource(KubernetesTestSupport.POD, NAME, NS, HTTP_NOT_FOUND);
+      testSupport.setTime(10, TimeUnit.SECONDS);
+
+      assertThat(terminalStep.wasRun(), is(true));
```

Comment:
And again, the test for `wasRun` is not needed, since you are checking that the step was run exactly once.

Reply:
Removed the `wasRun` test. Thanks.